src/poetry/core/factory.py

```diff
@@ -323,6 +324,25 @@ def _configure_package_metadata(
             )
             package.readmes = tuple(root / r for r in custom_readmes if r)
 
+        cls._validate_import_names(project)
+
+        package.import_names = project.get("import-names")
+        package.import_namespaces = project.get("import-namespaces")
+
+    @classmethod
+    def _validate_import_names(cls, project: dict[str, Any]) -> None:
+        import_names = {
+            name.split(";")[0].strip() for name in project.get("import-names", [])
+        }
+        import_namespaces = {
+            name.split(";")[0].strip() for name in project.get("import-namespaces", [])
+        }
+
+        if duplicates := import_names & import_namespaces:
+            raise PoetryCoreError(
+                f"Import names found in both import-names and import-namespaces: {', '.join(duplicates)}"
+            )
```

Comment:
This should not raise but add an error to the list (when properly called in `validate()`).